### PR TITLE
[CI-6246]: Add TI Support

### PR DESCRIPTION
**Pull request by rutvijmehta-harness**

### CODE REVIEW

File: handler/setup.go [hunk 1]
```diff
@@ -31,7 +33,7 @@ var random = func() string {
 	return uniuri.NewLen(20)
 }
 
-// HandleExecuteStep returns an http.HandlerFunc that executes a step
+// HandleSetup returns an http.HandlerFunc that executes a step
```
Comment (happyhebaby) [inline handler/setup.go]: Thanks for fixing the copied comment, the content should also be fixed as "runs an initialize step".

File: ti/callgraph/upload.go [hunk 2]
```diff
@@ -7,14 +7,14 @@ package callgraph
 import (
 	"context"
 	"fmt"
+	"github.com/harness/harness-docker-runner/api"
```
Comment (happyhebaby) [inline ti/callgraph/upload.go]: Nit, move harness imports to the right import group.

File: ti/report/report.go [hunk 3]
```diff
@@ -7,16 +7,17 @@ package report
...
+	"time"
 
 	"github.com/harness/harness-docker-runner/api"
-	"github.com/harness/harness-docker-runner/ti/client"
 	"github.com/harness/harness-docker-runner/ti/report/parser/junit"
 	"github.com/sirupsen/logrus"
 )
 
-func ParseAndUploadTests(ctx context.Context, report api.TestReport, workDir, stepID string, log *logrus.Logger, ticlient client.Client) error {
+func ParseAndUploadTests(ctx context.Context, report api.TestReport, workDir, stepID string, log *logrus.Logger, start time.Time, tiConfig *api.TIConfig) error {
```
Comment (happyhebaby) [inline ti/report/report.go]: I noticed some of the code in this method in lite-engine is skipped, is this intentional?

File: handler/setup.go (hunk 1 above)
Comment (rutvijmehta-harness) [reply handler/setup.go]: Done

File: ti/callgraph/upload.go (hunk 2 above)
Comment (rutvijmehta-harness) [reply ti/callgraph/upload.go]: Done

File: ti/report/report.go (hunk 3 above)
Comment (rutvijmehta-harness) [reply ti/report/report.go]: Thanks. It was removed during the code refactor. Added it back.

File: ti/callgraph/upload.go (hunk 2 above)
Comment (happyhebaby) [reply ti/callgraph/upload.go]: Hasn't seen the change, did you forget to push the change?

Comment (rutvijmehta-harness) [reply ti/callgraph/upload.go]: Done

**Review by hemanthmantri — APPROVED** [review]

**Pull request merged**